benchmarks/automated/tests/utils/benchmark.py

```diff
@@ -0,0 +1,382 @@
+                url = mode_config.get("url")
+                workers = mode_config.get("workers")
+                batch_delay = mode_config.get("batch_delay")
+                batch_sizes = mode_config.get("batch_size")
+                input_file = mode_config.get("input")
+                requests = mode_config.get("requests")
+                concurrency = mode_config.get("concurrency")
+                backend_profiling = mode_config.get("backend_profiling")
+                exec_env = mode_config.get("exec_env")
+                processors = mode_config.get("processors")
+
+                # values for workflow
+                workflow_name = mode_config.get("workflow_name")
+                workflow_model_urls = mode_config.get("models")
+                workflow_specfile_url = mode_config.get("specfile")
+                workflow_handler_url = mode_config.get("workflow_handler")
+                retry_attempts = mode_config.get("retry_attempts")
+                timeout_ms = mode_config.get("timeout_ms")
+
+                # url is just a name in this case
+                url = workflow_name
+
+                LOGGER.info(f"model_urls_in_workflow: {workflow_model_urls}")
+                LOGGER.info(f"workflow_specfile_url: {workflow_specfile_url}")
+                LOGGER.info(f"workflow_handler_url: {workflow_handler_url}")
+                LOGGER.info(f"workflow_name: {workflow_name}")
+
+                gpus = None
+                if len(processors) == 2:
+                    gpus = processors[1].get("gpus")
+                LOGGER.info(f"processors: {processors[1]}")
+                LOGGER.info(f"gpus: {gpus}")
+
+                LOGGER.info(
+                    f"\n benchmark_engine: {benchmark_engine}\n url: {url}\n workers: {workers}\n batch_delay: {batch_delay}\n batch_size:{batch_sizes}\n input_file: {input_file}\n requests: {requests}\n concurrency: {concurrency}\n backend_profiling: {backend_profiling}\n exec_env: {exec_env}\n processors: {processors}"
+                )
+
+                torchserveHandler = ts_utils.TorchServeHandler(
+                    exec_env=exec_env,
+                    cuda_version=cuda_version_for_instance,
+                    gpus=gpus,
+                    torchserve_docker_image=docker_repo_tag_for_current_instance,
+                    backend_profiling=backend_profiling,
+                    connection=self.connection,
+                    is_local_execution=self.is_local_execution
+                )
+
+                torchserveHandler.download_workflow_artifacts(
+                    workflow_name, workflow_model_urls, workflow_specfile_url, workflow_handler_url
+                )
+
+                for batch_size in batch_sizes:
+                    # Start torchserve
+                    torchserveHandler.start_torchserve_docker()
+
+                    # Create workflow archive and place in the wf_store
+                    torchserveHandler.create_and_update_workflow_archive(
```

Comment:
Why is create workflow archiver needed? why workflow model war file not in S3?

Reply:
The automated benchmark loops through different 'batch sizes', in case of a workflow, the batch-size is specified in the config yml within the workflow archive, and cannot be modified using REST APIs as of now, hence we need to create/re-create the archive for different executions of batch sizes.